types/base/bytes.go

```diff
@@ -49,6 +53,14 @@ func (this Bytes) Compare(that Bytes) int {
 	return bytes.Compare([]byte(this), []byte(that))
 }
 
+func (this Bytes) Equal(that Bytes) bool {
```

Comment:
receiver name should be a reflection of its identity; don't use generic names such as "this" or "self"